Frameworks/AddressBook/ABRecord.mm

```diff
@@ -47,12 +82,118 @@ bool ABRecordSetValue(ABRecordRef record, ABPropertyID property, CFTypeRef value
+    } else if (contactProperty == kABPersonMiddleNameProperty) {
+        return (__bridge_retained CFTypeRef)person.contact.middleName;
+    } else if (contactProperty == kABPersonPrefixProperty) {
+        return (__bridge_retained CFTypeRef)person.contact.honorificNamePrefix;
+    } else if (contactProperty == kABPersonSuffixProperty) {
+        return (__bridge_retained CFTypeRef)person.contact.honorificNameSuffix;
+    } else if (contactProperty == kABPersonNicknameProperty) {
+        return (__bridge_retained CFTypeRef)person.contact.nickname;
+    } else if (contactProperty == kABPersonFirstNamePhoneticProperty) {
+        return (__bridge_retained CFTypeRef)person.contact.yomiGivenName;
+    } else if (contactProperty == kABPersonLastNamePhoneticProperty) {
+        return (__bridge_retained CFTypeRef)person.contact.yomiFamilyName;
+    }
+
+    // Cases for job-related properties.
+    if (contactProperty == kABPersonOrganizationProperty) {
+        NSArray* jobInfo = person.contact.jobInfo;
+        return [jobInfo count] > 0 ? (__bridge_retained CFTypeRef)((WACContactJobInfo*)jobInfo[0]).companyName : nullptr;
+    } else if (contactProperty == kABPersonJobTitleProperty) {
+        NSArray* jobInfo = person.contact.jobInfo;
+        return [jobInfo count] > 0 ? (__bridge_retained CFTypeRef)((WACContactJobInfo*)jobInfo[0]).title : nullptr;
+    } else if (contactProperty == kABPersonDepartmentProperty) {
+        NSArray* jobInfo = person.contact.jobInfo;
+        return [jobInfo count] > 0 ? (__bridge_retained CFTypeRef)((WACContactJobInfo*)jobInfo[0]).department : nullptr;
+    }
+
+    // Case for birthday-related property.
+    if (contactProperty == kABPersonBirthdayProperty) {
+        NSArray* dates = person.contact.importantDates;
+        NSUInteger birthdayIndex = [dates indexOfObjectPassingTest:^BOOL(id obj, NSUInteger idx, BOOL* stop) {
+            WACContactDate* date = (WACContactDate*)obj;
+            if (date.kind == WACContactDateKindBirthday) {
+                *stop = YES;
+                return YES;
+            } else {
+                return NO;
+            }
+        }];
+
+        if (birthdayIndex == NSNotFound) {
+            return nullptr;
+        } else {
+            WACContactDate* date = dates[birthdayIndex];
+            NSCalendar* calendar = [NSCalendar currentCalendar];
+            NSDateComponents* dateComponents = [[NSDateComponents alloc] init];
+            dateComponents.day = [date.day integerValue];
+            dateComponents.month = [date.month integerValue];
+            dateComponents.year = [date.year integerValue];
+            NSDate* resultDate = [calendar dateFromComponents:dateComponents];
+            return (__bridge_retained CFTypeRef)resultDate;
+        }
+    }
+
+    // Case for note-related property.
+    if (contactProperty == kABPersonNoteProperty) {
+        return (__bridge_retained CFTypeRef)person.contact.notes;
```

Comment:
The notes are stored in a String (in both iOS and Windows)

---
In reply to: [74828015](https://github.com/Microsoft/WinObjC/pull/881#discussion_r74828015) [](ancestors = 74828015)